apis/slo/v1alpha1/nodemetric_types.go

```diff
@@ -23,8 +23,25 @@ import (
 // EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
 // NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
 
+type AggregationType string
+
+const (
+	// max is not welcomed since it may import outliers
+	AVG    AggregationType = "avg"
+	P99    AggregationType = "p99"
+	P95    AggregationType = "p95"
+	P90    AggregationType = "p90"
+	MEDIAN AggregationType = "median"
+)
+
 type NodeMetricInfo struct {
-	NodeUsage ResourceMap `json:"nodeUsage,omitempty"`
+	NodeUsage            ResourceMap       `json:"nodeUsage,omitempty"`
```

Comment:
If the field is optional, it had better be a pointer type like `*ResourceMap`, or we can remove the `omitempty` to make it a required field.

Reply:
it is an internal protocol, and ResourceMap contains an inline map. so it does not need to be defined as pointer type.